MultiPoolMiner.ps1

```diff
@@ -160,11 +162,21 @@ $API.Version = $Version
 while ($true) {
     $ConfigBackup = $Config
     #Load the config, read command line parameters except ConfigFile, use default values for those that are not in command line
-    $Parameters = @{}
+    if ($PSBoundParameters.ConfigFile) {
```

Comment:
$PSBoundParameters only gets set when a valid is explicitly given, if it's just the default valid $PSBoundParameters will be null.  This means that the default of Config.txt will be completely ignored and no config file loaded by default.

Same goes for generating the default config file above.  It will never automatically create the config.txt file unless you specify '-configfile config.txt' in the command line.

If there is no -configfile parameter, it should use the default of config.txt if it exists (or create it if it doesn't), so you can't use $PSBoundParameters.

Reply:
> it should use the default of config.txt 

Nope, that's the point. MPM does not need a config file. But you can use one if you want.
That's why I code it like this. If -ConfigFile is omitted then just use the 'in-memory' config created in lines 172 -179.

Reply:
I disagree on this one.  I prefer being able to start multipoolminer.ps1 without any parameters, and have the config file take care of all the settings.

I do however agree that if a setting is missing from config.txt but present in the command line, it should use what's in the command line (which it currently doesn't), and if config file and command line parameters conflict, the command line parameter should take priority.  Those changes I like.